packages/core/src/model-types.ts

```diff
@@ -0,0 +1,52 @@
+type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);
+
+/**
+ * Azure OpenAI models for chat completion.
+ */
+export type AzureOpenAiChatModel = LiteralUnion<
```

Comment:
I think this is a good idea. The names will be even longer through that, but theoretically we could also provide a second client for OpenAI only, right? @deekshas8 also interested in your opinion.

Reply:
I also like the idea of adding `Azure`, and if we do it, would also want to be adapted in the Java SDK as well.
But, do you know if there are any differences between the `AzureOpenAI` and  plain `OpenAI` clients?

I wasn't aware that Anthropic Claude is served by AWS. Where did you read about this?

Reply:
@KavithaSiva https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/models-and-scenarios-in-generative-ai-hub

Reply:
I not sure if having a client for pure OpenAI would be possible because that would require AI Core to support it first. But Anyway, `AzureOpenAI` is how the model is called in AI Core deployment as well as in the [SAP Help documentation](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-deployment-for-generative-ai-model-in-sap-ai-core#using-the-api).

Regarding Anthropic Claude, the executable id of all `anthropic--claude-*` models is `aws-bedrock`. So it is provided via AWS. You can also find model details in the yaml file I mentioned in the backlog item.